### ISSUE TITLE
[Project - Onboarding]: Implement Tooltip functionality (without styling)

### ISSUE THREAD

**Issue by valentinpalkovic** [storybookjs/storybook#22673]
# Implement React Component: Tooltip

## Objective:
The objective of this issue is to create a new React component named `Tooltip`. As discussed in https://github.com/storybookjs/storybook/issues/22667, use [Floating UI](https://floating-ui.com/) to create the tooltip!

### 1. Properties:

| Property  | Type | Description | Required |
| --------- | ---- | ----------- | -------- |
| children  | React.ReactNode | Children elements. | Yes |
| anchorElement | Element | The HTMLElement reference, on which the tooltip should be placed on. | Yes |
| ...tooltipProps  | TooltipProps | Tooltip properties for fine-grained control. | No |

### 2. Behaviors:

- The tooltip visibility is controlled outside. The parent component decides whether the tooltip should be displayed or not.
- Use `React.Portal` to display the tooltip at `document.body` to guarantee a proper overlay over the underlying content

### 3. Design (DRAFT):
![image](https://github.com/storybookjs/storybook/assets/5889929/17aa2eec-8904-4ef1-8da7-34c60c4b78d9)
![image](https://github.com/storybookjs/storybook/assets/5889929/687fa334-f881-4359-87a9-1b17567e4ab1)

### 4. Blocked by:

```[tasklist]
- [ ] https://github.com/storybookjs/storybook/issues/22667
```

## Acceptance Criteria:

- The component is implemented according to the requirements described above, or an existing component is taken that fulfills the requirements.
- The component passes all defined test cases and has Stories displaying its states
- The component linked to a figma component in Storybook - speak to designers!
- You should not implement any complex CSS styles! The design is still a work in progress. The component will be finalized design-wise in a separate ticket. Make sure that the `properties` and the `behaviors` are implemented accordingly.
- Darkening the background or highlighting the reference HTMLElement is not part of this ticket.
